src/beast/beast/wsproto/detail/mask.h

```diff
@@ -0,0 +1,396 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of Beast: https://github.com/vinniefalco/Beast
+    Copyright 2013, Vinnie Falco <vinnie.falco@gmail.com>
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef BEAST_WSPROTO_DETAIL_MASKGEN_H_INCLUDED
+#define BEAST_WSPROTO_DETAIL_MASKGEN_H_INCLUDED
+
+#include <boost/asio/buffer.hpp>
+#include <array>
+#include <climits>
+#include <cstdint>
+#include <random>
+#include <type_traits>
+
+namespace beast {
+namespace wsproto {
+namespace detail {
+
+// Pseudo-random source of mask keys
+//
+template<class = void>
+class maskgen_t
+{
+private:
+    std::size_t n_;
+    std::mt19937 g_;
+
+public:
+    using result_type = typename std::mt19937::result_type;
+
+    maskgen_t(maskgen_t const&) = delete;
+    maskgen_t& operator=(maskgen_t const&) = delete;
+
+    maskgen_t();
+
+    result_type
+    operator()() noexcept;
+
+    void
+    rekey();
+};
+
+template<class _>
+maskgen_t<_>::maskgen_t()
+{
+    rekey();
+}
+
+template<class _>
+auto
+maskgen_t<_>::operator()() noexcept ->
+    result_type
+{
+    if(++n_ == 10000)
```

Comment:
10000 is awfully low here.  This probably doesn't help add entropy and may well hurt.  The period of `std::mt19937` is 2^19937 − 1 (https://en.wikipedia.org/wiki/Mersenne_Twister).